extension/chrome/settings/modules/contacts.ts

```diff
@@ -108,8 +108,8 @@ View.run(class ContactsView extends View {
     }
   }
 
-  private actionExportAllKeysHandler = () => {
-    const allArmoredPublicKeys = this.contacts.map(c => c.armoredPubkey).filter(Boolean).map(a => a!.trim()).join('\n');
+  private actionExportAllKeysHandler = async () => {
+    const allArmoredPublicKeys = (await ContactStore.searchPubkeys(undefined, { has_pgp: true, limit: 500, substring: String($('.input-search-contacts').val()) })).map(a => a!.trim()).join('\n');
```

Comment:
Wow - this was a bug before, and it's still a bug now. I'm glad you highlighted this!

This functionality is supposed to export every single public key that's in the database, not the currently displayed ones. And there should be no limit.

Reply:
Done. Should I create a test for it?

Reply:
It's a bit harder to test, you'd have to import > 500 keys to see the difference. Thank you for asking - let's skip test this time.